palace/drivers/electrostaticsolver.cpp

```diff
@@ -65,19 +72,24 @@ ElectrostaticSolver::Solve(const std::vector<std::unique_ptr<Mesh>> &mesh) const
                linalg::Norml2(laplaceop.GetComm(), V[step]),
                linalg::Norml2(laplaceop.GetComm(), RHS));
 
+    // Calculate and record the error indicators.
+    Mpi::Print(" Updating solution error estimates\n");
+    estimator.AddErrorIndicator(V[step], indicator);
```

Comment:
The original design had the estimation as part of the main loop, but I moved it to post processing at your request, why the move back?

Reply:
When the estimation is fast, it is nice to have at the end I think since the solution vectors are stored. However when it's slow, this leads to the code appearing to "hang" at the end of the simulation when the error estimates are calculated for all vectors which is remedied when they are updated one at a time in the loop. Also, if the error estimation is going to lead to an out of memory error, it's nice to run into that earlier rather than only after all the solves have occurred. That's my thinking.